src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java

```diff
@@ -365,4 +366,68 @@ void shouldThrowAnExceptionWhenWalletIdEqualsZero() throws Exception {
         //then
         result.andExpect(status().isNoContent());
     }
+
+    @Test
+    void shouldReturnStatusOkWhenFindWalletByIdCorrectly() throws Exception {
+        // given
+        Instant creationDate = Instant.now();
+        WalletDTO wallet = new WalletDTO(1L, "Default", creationDate);
+
+        // when
+        when(walletService.findById(1L)).thenReturn(wallet);
+        ResultActions result = mockMvc.perform(get("/api/wallet/{id}", 1L)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(Objects.requireNonNull(objectMapper.writeValueAsString(wallet))));
+
+        // then
+        result.andExpect(status().isOk());
+    }
+
+    @Test
+    void shouldReturnCorrectnessOfDataInFoundWalletWhenFindWalletById() throws Exception {
+        // given
+        Instant creationDate = Instant.now();
+        WalletDTO wallet = new WalletDTO(1L, "Default", creationDate);
+
+        // when
+        when(walletService.findById(1L)).thenReturn(wallet);
+        ResultActions result = mockMvc.perform(get("/api/wallet/{id}", 1L)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(Objects.requireNonNull(objectMapper.writeValueAsString(wallet))));
+
+        // then
+        result.andExpect(jsonPath("$.id").value(1L))
+                .andExpect(jsonPath("$.name").value("Default"))
+                .andExpect(jsonPath("$.creationDate").value(creationDate.toString()));
+    }
+
+    @Test
+    void shouldThrowWalletNotFoundExceptionWhenWalletByIdDoesNotExist() throws Exception {
```

Comment:
you called the method as a ``shouldThrowWalletNotFoundExceptionWhenWalletByIdDoesNotExist()``, but you are checking if the response status is equal to Not Found.